### DISCUSSION TITLE
Production Infra setup for Cokreat

### DISCUSSION THREAD

**Discussion by tsprasath** [Sunbird-coKreat/Community#42] [General Discussions]
Hi Team,

Need to setup cokreat for production. kindly direct us if any document or steps to follow

**Comment by arun1205** [comment]
@gandham-santhosh Can u please look into this

**Reply by vinukumar-vs** [reply]
@tsprasath @arun1205 I have discussed with @gandham-santhosh, We dont have the document for server installation of CoKreat BB. I will update it at the earliest.

For your reference please follow the below architecture diagram's 
<h3>Deployment Diagram</h3>

https://cokreat.sunbird.org/use/overview


<h3> Dependency Diagram of CoKreat</h3>

https://cokreat.sunbird.org/learn/technical-architecture


<h3> BB dependencies </h3>

https://cokreat.sunbird.org/use/learn-more/dependencies

**Reply by vinukumar-vs** [reply]
@arun1205 @tsprasath We had an internal discussion on this. We will be sharing the first version of the document by 27/01/23.
@gandham-santhosh

**Reply by Krishnaj20** [reply]
@vinukumar-vs @mmanimegalai - is the doc shared?

**Reply by vinukumar-vs** [reply]
@tsprasath Please take a look at the below installation document for CoKreat installation.
https://cokreat.sunbird.org/use/installation-guide/installation-document

@gandham-santhosh @Krishnaj20 @mmanimegalai

**Reply by tsprasath** [reply]
Hi @vinukumar-vs @Krishnaj20  Followed the document provided by you. 

I was able to complete full cycle of installation, Access via https://34.93.40.45.

However the UI is pointing to Portal and not the cokreat or vidyadaan.

I tried to figure out the issue but could not corner it.

It would be great if you can help me in resolving the issue.

![Screenshot from 2023-02-09 21-13-58](https://user-images.githubusercontent.com/20708426/217863596-0b0ea654-c7e6-43f4-a370-3d920b227fc4.png)

**Reply by Krishnaj20** [reply]
@vinukumar-vs @arun1205 - Pls assist here.

**Reply by vinukumar-vs** [reply]
@tsprasath Sunbird-ED(portal) & CoKreat(portal) are completely different infra itself. 
You should have different host for Sunbird-ED & Cokreat.

If the below link is pointing to CoKreat infra, it should load CoKreat(portal) or nothing(Because sunbird-ed portal itself is not available or deployed on CoKreat infra).

> I was able to complete full cycle of installation, Access via https://34.93.40.45/.

@santhosh-tg Can you verify the [installation document](https://cokreat.sunbird.org/use/installation-guide/installation-document) & help @tsprasath to have different hosts of Sunbird-ED & CoKreate portal? 


To access CoKoreat portal, refer the below links
{{host}}//sourcing (To create new programs)
{{host}}/contribute (To contribute to program)

Tagging @gandham-santhosh

**Reply by tsprasath** [reply]
@vinukumar-vs I have seen the cokreat portal referred to the repo https://github.com/Sunbird-Ed/creation-portal. 

In the document not able to find the above repo

**Reply by santhosh-tg** [reply]
In the documentation  Player  is pointing to https://github.com/Sunbird-Ed/SunbirdEd-portal.git @tsprasath  Please build the Player service from repo https://github.com/Sunbird-Ed/creation-portal and deploy. We will update the documentation  to reflect it

**Reply by Krishnaj20** [reply]
@tsprasath - is the issue resolved? 

@santhosh-tg @gandham-santhosh

**Reply by Krishnaj20** [reply]
@tsprasath - Please confirm if the issue is now resolved?

**Reply by vinukumar-vs** [reply]
@tsprasath I have updated the documentation.. 
https://cokreat.sunbird.org/use/installation-guide/installation-document#build-and-deploy-services
Build/job/DockCore/player
Build/job/DockCore/Program

We will confirm the dependency on (may not require)
Build/Core/Player => Handles the UI elements => https://github.com/Sunbird-Ed/SunbirdEd-portal.git

**Reply by tsprasath** [reply]
After deploying the right repo our UI to points to vidyadaan.

**Reply by Krishnaj20** [reply]
thanks @tsprasath . Locking this conversation as the issue has been resolved.
